reactjs-films-homework/src/components/Header/Form/Form.tsx

```diff
@@ -1,29 +1,86 @@
-import React, { useRef } from 'react';
+import React, { useState } from 'react';
 import search from './assets/search.png';
 import styles from './Form.module.scss';
-const Form: React.FC = () => {
-  const placeholder = 'Movies, person, movie theaters';
-  const ref: any = useRef();
-  const loc = () => {
-    let inputValue = ref.current.value;
-    inputValue ? console.log(inputValue) : console.log('No matches');
-    inputValue = '';
+import { useAppSelector, useAppDispatch } from '../../../hooks/hooks';
+import {
+  setQuery,
+  setActivePage,
+} from '../../../store/rootStore/mainStore/mainSlice';
+import getMainData from '../../../store/rootStore/mainStore/getMaiData/getMainData';
+
+type IFormProps = {
+  placeholder: string;
+};
+
+const Form: React.FC<IFormProps> = ({ placeholder }) => {
+  const [inputValue, setInputValue] = useState<string>('');
+  const pageAfterSearch = '1';
+  const dispatch = useAppDispatch();
+  const [placeholderState, setPlaceholderState] = useState<string>(placeholder);
+  const mainState = {
+    lang: useAppSelector((state) => state.mainReducer.lang),
+    category: useAppSelector((state) => state.mainReducer.category),
+    page: pageAfterSearch,
+  };
+
+  const resetInput = () => {
+    const checkIsOnBlur = document.body.addEventListener(
+      'click',
+      (event: any) => {
+        const id = event.target.id;
+        if (id === 'search') {
+          setInputValue('');
+        }
+        if (id === 'input_search') {
+          return;
+        }
+        setInputValue('');
+      },
+      { once: true }
+    );
+    return checkIsOnBlur;
+  };
+
+  const getSearchResults = () => {
+    setPlaceholderState((prev) => {
+      return (prev = !inputValue ? placeholder : `Results for ${inputValue}`);
+    });
+    dispatch(setActivePage(pageAfterSearch));
+    dispatch(setQuery(inputValue));
+    dispatch(getMainData({ ...mainState, query: inputValue }));
   };
 
   return (
-    <div className={styles.formWrapper}>
-      <form className={styles.form}>
+    <div className={styles.formWrapper} onBlur={resetInput}>
+      <form className={styles.form} id={'form'}>
         <input
-          defaultValue=""
-          ref={ref}
           type="search"
           className={styles.input}
-          placeholder={placeholder}
+          placeholder={placeholderState}
           id="input_search"
           autoComplete="off"
+          onChange={(event) => {
+            setInputValue((prev) => (prev = event.target.value));
+          }}
```

Comment:
It's good practice to move such handlers to separate 'const' for better readability.